openbb_platform/providers/bls/openbb_bls/utils/helpers.py

```diff
@@ -0,0 +1,424 @@
+            expanded_data = []
+            for item in ids["footnote_code"]:
+                if (
+                    item
+                    and isinstance(item, str)
+                    and "," in item
+                    and any(char.isdigit() for char in item)
+                ):
+                    expanded_data.append(
+                        " ".join(
+                            [
+                                code_map["footnote_code"].get(sub_item, sub_item)
+                                for sub_item in item.split(",")
+                                if code_map["footnote_code"].get(sub_item) is not None
+                            ]
+                        )
+                    )
+                else:
+                    expanded_data.append(item)
+            ids["footnote_code"] = expanded_data
+
+        ids = ids.replace({nan: None, "nan": None, "''": None}).to_dict(
+            orient="records"
+        )
+        series_ids.extend(ids)
+        series_codes.update({survey: code_map})
+
+    # Iterate over the all the surveys in the category and generate the tables and code maps.
+    # The FTP doesn't seem to like a flood of requests, so we do this operation in series.
+    for survey in SURVEY_CATEGORY_MAP[category]:
+        await get_all_series_ids(survey.lower())
+
+    return series_ids, series_codes
+
+
+def open_asset(asset: str) -> Union["DataFrame", Dict]:
+    """Open a static file asset for series IDs or code maps."""
+    # pylint: disable=import-outside-toplevel
+    import os  # noqa
+    import json
+    from importlib.resources import files
+    from pathlib import Path
+    from numpy import nan
+    from openbb_core.app.model.abstract.error import OpenBBError
+    from pandas import read_csv
+
+    if ".xz" not in asset and "series" in asset:
+        asset = asset + ".xz"
+    elif ".json" not in asset and "codes" in asset:
+        asset = asset + ".json"
+    elif ".json" in asset or ".xz" in asset:
+        pass
+    else:
+        raise OpenBBError(f"Asset '{asset}' not supported. Expected .json or .xz file.")
+
+    assets_path = Path(str(files("openbb_bls").joinpath("assets")))
+
+    if not os.path.exists(assets_path.joinpath(asset)):
+        raise OpenBBError(f"Asset '{asset}' not found.")
+
+    if asset.endswith(".json"):
+        with open(assets_path.joinpath(asset)) as f:
+            return json.load(f)
+    else:
+        with open(assets_path.joinpath(asset), "rb") as f:
+            df = read_csv(f, compression="xz", low_memory=False, dtype="str")
+        return df.replace({nan: None, "nan": None, "''": None}).dropna(
+            how="all", axis=1
+        )
+
+
+async def update_static_asset(category: str) -> None:
+    """Update a static file assets with series IDs and code maps for a given category.
+    Do not use unless the static files in the assets folder require updating.
+    """
+    # pylint: disable=import-outside-toplevel
```

Comment:
Cant you just do this at the top once instead of inside each function

Reply:
Yes, I suppose so, the intention of keeping it there is to be able to more easily identify/track the specific locations where this occurs.

That would be a bigger refactor touching nearly every file in the repo.